x/liquidstaking/keeper/net_amount.go

```diff
@@ -55,11 +55,20 @@ func (k Keeper) GetNetAmountState(ctx sdk.Context) (nas types.NetAmountState) {
 			cachedCtx, _ := ctx.CacheContext()
 			endingPeriod := k.distributionKeeper.IncrementValidatorPeriod(cachedCtx, validator)
 			delRewards := k.distributionKeeper.CalculateDelegationRewards(cachedCtx, validator, delegation, endingPeriod)
+			// chunk's remaining reward is calculated by
+			// 1. rest = del_reward - insurance_commission
+			// 2. remaining = rest x (1 - module_fee_rate)
+			// 3. result = remaining x (1 - discount_rate)
 			delReward := delRewards.AmountOf(bondDenom)
 			insuranceCommission := delReward.Mul(pairedIns.FeeRate)
-			// insuranceCommission is not reward of module
-			pureReward := delReward.Sub(insuranceCommission)
-			totalRemainingRewards = totalRemainingRewards.Add(pureReward)
+			restReward := delReward.Sub(insuranceCommission)
+			moduleFeeRate, _ := k.CalcDynamicFeeRate(ctx)
+			remainingReward := restReward.Mul(sdk.OneDec().Sub(moduleFeeRate))
+			totalRemainingRewards = totalRemainingRewards.Add(
+				remainingReward.Mul(
+					sdk.OneDec().Sub(k.CalcDiscountRate(ctx)),
```

Comment:
I think we don't need to re-calc `CalcDynamicFeeRate` and `CalcDiscountRate` for every chunk

after all, we set `nas.FeeRate, nas.UtilizationRatio = k.CalcDynamicFeeRate(ctx)`, `nas.DiscountRate = k.CalcDiscountRate(ctx)`

If we set `nas.FeeRate, nas.UtilizationRatio,  nas.DiscountRate`  in advance and recycle it when calculating `delReward`, rather than calculating it every iteration, the operation will be greatly reduced

Reply:
Thanks for the feedback.
Optimized it in https://github.com/b-harvest/Canto/pull/18/commits/5c8ee8cca24ecb600a7c78acdc61815b590643e3

Reply:
need to replace duplicated `CalcDynamicFeeRate`, `CalcDiscountRate` call on [here](https://github.com/b-harvest/Canto/commit/5c8ee8cca24ecb600a7c78acdc61815b590643e3#diff-04fc6a1fc66091b4accc176f864244fdc95aac78a227371bc97e8e40814dbc63R117-R119)

Reply:
done! thanks.
https://github.com/b-harvest/Canto/pull/18/commits/3255774f8eb48a35c6554cc31d62020a47781e9f